operator/pkg/gateway-api/httproute_gamma_reconcile.go

```diff
@@ -0,0 +1,203 @@
+		return controllerruntime.Fail(err)
+	}
+
+	// Ignore deleted HTTPRoute, this can happen when foregroundDeletion is enabled
+	if original.GetDeletionTimestamp() != nil {
+		return controllerruntime.Success()
+	}
+
+	hr := original.DeepCopy()
+
+	// Get ReferenceGrants
+	grants := &gatewayv1beta1.ReferenceGrantList{}
+	if err := r.Client.List(ctx, grants); err != nil {
+		return r.handleReconcileErrorWithStatus(ctx, fmt.Errorf("failed to retrieve reference grants: %w", err), original, hr)
+	}
+
+	servicesList := &corev1.ServiceList{}
+	if err := r.Client.List(ctx, servicesList); err != nil {
+		scopedLog.WithError(err).Error("Unable to list Services")
+		return r.handleReconcileErrorWithStatus(ctx, err, original, hr)
+	}
+
+	// input for the validators
+	i := &routechecks.HTTPRouteInput{
+		Ctx:       ctx,
+		Logger:    scopedLog.WithField(logfields.Resource, hr),
+		Client:    r.Client,
+		Grants:    grants,
+		HTTPRoute: hr,
+	}
+
+	// gateway validators
+	for _, parent := range hr.Spec.ParentRefs {
+
+		// set acceptance to okay, this wil be overwritten in checks if needed
+		i.SetParentCondition(parent, metav1.Condition{
+			Type:    string(gatewayv1.RouteConditionAccepted),
+			Status:  metav1.ConditionTrue,
+			Reason:  string(gatewayv1.RouteReasonAccepted),
+			Message: "Accepted HTTPRoute",
+		})
+
+		// set status to okay, this wil be overwritten in checks if needed
+		i.SetAllParentCondition(metav1.Condition{
+			Type:    string(gatewayv1.RouteConditionResolvedRefs),
+			Status:  metav1.ConditionTrue,
+			Reason:  string(gatewayv1.RouteReasonResolvedRefs),
+			Message: "Service reference is valid",
+		})
+
+		for _, fn := range []routechecks.CheckParentFunc{
+			routechecks.CheckGammaServiceAllowedForNamespace,
+		} {
+			continueCheck, err := fn(i, parent)
+			if err != nil {
+				return r.handleReconcileErrorWithStatus(ctx, fmt.Errorf("failed to apply Gateway check: %w", err), original, hr)
+			}
+
+			if !continueCheck {
+				break
+			}
+		}
+	}
+
+	for _, fn := range []routechecks.CheckRuleFunc{
+		routechecks.CheckAgainstCrossNamespaceBackendReferences,
+		routechecks.CheckBackend,
+		routechecks.CheckBackendIsExistingService,
+	} {
+		continueCheck, err := fn(i)
+		if err != nil {
+			return r.handleReconcileErrorWithStatus(ctx, fmt.Errorf("failed to apply Backend check: %w", err), original, hr)
+		}
+
+		if !continueCheck {
+			break
+		}
+	}
+
+	if err := r.updateStatus(ctx, original, hr); err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to update HTTPRoute status: %w", err)
+	}
+
+	httpListeners := ingestion.GammaHTTPRoutes(ingestion.GammaInput{
+		HTTPRoutes:      []gatewayv1.HTTPRoute{*hr},
+		Services:        servicesList.Items,
+		ReferenceGrants: grants.Items,
+	})
+
+	cec, svc, cep, err := r.translator.Translate(&model.Model{HTTP: httpListeners})
+	if err != nil {
+		scopedLog.WithError(err).Error("Unable to translate resources")
+		// setGatewayAccepted(gw, false, "Unable to translate resources")
```

Comment:
Is this commented out by accident?

Reply:
No, we don't interact with Gateways, so we can't set them to accepted. Should be removed though. I'll do that now, thanks!